scrapperFunctions/index.js

```diff
@@ -0,0 +1,227 @@
+    for (let j = 0; j < elements.length; j++) {
+      if (el[j].innerText == '' || el[j].innerText == 'undefined') {
+        indArr.push('---');
+      } else {
+        indArr.push(el[j].innerText);
+      }
+    }
+    inf.push(indArr);
+  }
+  return inf;
+  // eslint-disable-next-line comma-dangle
+};
+
+/**
+ * Get the number of rows in the table
+ * @param {string} selector Of where I'm going to get the data
+ * @returns {int | string} The data obtained
+ */
+
+const getRows = (selector) => {
+  const rows = document.querySelector(selector).innerText;
+  return rows;
+};
+
+/**
+ * How many rows there is
+ * @param {string} select How identify the element
+ * @returns {int | string} The data obtained
+ */
+// eslint-disable-next-line space-before-function-paren
+ScrapPage.prototype.expectedRows = async function (select) {
+  try {
+    const rowsQ = await this.page.evaluate(getRows, select);
+    return rowsQ;
+  } catch (err) {
+    return err;
+  }
+};
+
+/**
+ * Collect the data of different screens
+ * @param {string} tableSelector The identifier of the table
+ * @param {string} rowSelector How can I find the number of rows in the table
+ * @param {string} nextPageButton The identifier of the button
+ * @param {int | double} time How much time wait for do the process again
+ * @param {function} callback A function needed in progressBar
+ * @returns Array[Array[string]]
+ */
+// eslint-disable-next-line space-before-function-paren
+ScrapPage.prototype.checkData = async function (
+  tableSelector,
+  rowSelector,
+  nextPageButton,
+  time,
+  // eslint-disable-next-line prettier/prettier
+  callback,
+) {
+  try {
+    let exp = await this.expectedRows(rowSelector);
+    let data = await this.getDataTable(tableSelector);
+    const validateNumber = /^[0-9]*$/;
+    const onlyNumbers = validateNumber.test(exp);
+    typeof exp !== 'string' && onlyNumbers ? (exp = data.length) : exp;
+    let obt = data.length;
+    if (exp == 0) return exp;
+    console.log(`Expected data: ${exp}`);
+    console.log(`Getting data...`);
+
+    while (exp > obt) {
+      await this.progressBar(data.length, exp, callback);
+      await this.clickButton(nextPageButton, time);
+      const newData = await this.getDataTable(tableSelector);
+      data = data.concat(newData);
+      obt += newData.length;
+    }
+    await this.progressBar(data.length, exp, callback);
+    return data;
+  } catch (err) {
+    console.error(`Error: ${err}`);
+    throw err;
+  }
+};
+
+/**
+ * A function to know the percentage of data obtained
+ * @param {int} data The quantity of data obtained at this moment
+ * @param {int} expected How much data should be
+ * @param {function} callback How print thw data
+ */
+// eslint-disable-next-line space-before-function-paren
+ScrapPage.prototype.progressBar = function (data, expected, callback) {
+  try {
+    const percentage = Math.round((100 * data) / expected);
+    callback(percentage);
```

Comment:
I'd say, you dont need a callback for this, just add the print function code here